logsmanagement/parser.c

```diff
@@ -0,0 +1,1622 @@
+                goto next_item;
+            }
+
+            // TODO: Add below case in code!!!
+            // nginx $host and $http_host return ipv6 in [], apache doesn't
+            // TODO: TEST! This case hasn't been tested!
+            // char *pch = strchr(parsed[i], ']');
+            // if(pch){
+            //     *pch = '\0';
+            //     memmove(parsed[i], parsed[i]+1, strlen(parsed[i]));
+            // }
+
+            // field_size = field_size > VHOST_MAX_LEN ? VHOST_MAX_LEN : field_size;
+            snprintf( log_line_parsed->vhost, 
+                      field_size > (VHOST_MAX_LEN - 1) ? VHOST_MAX_LEN : (field_size + 1), 
+                      "%.*s", (int) field_size, field);
+
+            if(verify){
+                // if(field_size >= VHOST_MAX_LEN){
+                //     #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                //     error("VHOST is invalid");
+                //     #endif
+                //     log_line_parsed->vhost[0] = '\0';
+                //     log_line_parsed->parsing_errors++;
+                //     goto next_item; // TODO: Not entirely right, as it will also skip PORT parsing in case of VHOST_WITH_PORT
+                // }
+                
+                if(unlikely(regexec(&vhost_regex, log_line_parsed->vhost, 0, NULL, 0) == REG_NOMATCH)){
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("VHOST is invalid");
+                    #endif
+                    // log_line_parsed->vhost[0] = 'invalid';
+                    snprintf(log_line_parsed->vhost, sizeof(WEB_LOG_INVALID_HOST_STR), WEB_LOG_INVALID_HOST_STR);
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Extracted VHOST:%s", log_line_parsed->vhost);
+            #endif
+
+            if(fields_format[i] == VHOST) goto next_item;
+        }
+
+        if(fields_format[i] == VHOST_WITH_PORT || fields_format[i] == PORT){
+
+            if(fields_format[i] != VHOST_WITH_PORT){
+                port = field;
+                port_size = field_size;
+            }
+
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Item %d (type: PORT):%.*s", i, (int) port_size, port);
+            #endif
+
+            if(unlikely(port[0] ==  '-' && port_size == 1)){
+                log_line_parsed->port = WEB_LOG_INVALID_PORT;
+                log_line_parsed->parsing_errors++;
+                goto next_item;
+            }
+
+            char port_d[PORT_MAX_LEN];
+            snprintf( port_d, 
+                      port_size > (PORT_MAX_LEN - 1) ? PORT_MAX_LEN : (port_size + 1), 
+                      "%.*s", (int) port_size, port);
+
+            if(likely(str2int(&log_line_parsed->port, port_d, 10) == STR2XX_SUCCESS)){
+                if(verify){
+                    if(unlikely(log_line_parsed->port < 80 || log_line_parsed->port > 49151)){
```

Comment:
Port range can be set for values different of these limits you set, why are not accepting port smaller than 80 and higher than 49151?